src/SayMore/Model/ArchivingHelper.cs

```diff
@@ -449,14 +454,36 @@ private static void AddIMDIProjectData(Project saymoreProject, ArchivingDlgViewM
 					var language = LanguageList.FindByISO3Code(parts[0]);
 
 					// SP-765:  Allow codes from Ethnologue that are not in the Arbil list
-					if (string.IsNullOrEmpty(language?.EnglishName))
-						_defaultLanguage = new ArchivingLanguage(parts[0], parts[1], parts[1]);
-					else
-						_defaultLanguage = new ArchivingLanguage(language.Iso3Code, parts[1], language.EnglishName);
+					_defaultLanguage = string.IsNullOrEmpty(language?.EnglishName) ?
+						new ArchivingLanguage(parts[0], parts[1], parts[1]) :
+						new ArchivingLanguage(language.Iso3Code, parts[1], language.EnglishName);
 					package.ContentIso3Languages.Add(_defaultLanguage);
 				}
 			}
 
+			// analysis language
+			if (!string.IsNullOrEmpty(saymoreProject.AnalysisISO3CodeAndName))
+			{
+				var parts = saymoreProject.AnalysisISO3CodeAndName.SplitTrimmed(':').ToArray();
```

Comment:
I see two sections now doing similar parsing of ISO3CodeAndName consider refactoring the logic into a method that can be used in both places.